crates/control-interface/src/types/link.rs

```diff
@@ -7,11 +7,16 @@ use crate::{
 };
 
 /// A link definition between a source and target component (component or provider) on a given
-/// interface. An [`InterfaceLinkDefinition`] connects one component's import to another
+/// interface.
+///
+/// An [`InterfaceLinkDefinition`] connects one component's import to another
 /// component's export, specifying the configuration each component needs in order to execute
 /// the request, and represents an operator's intent to allow the source to invoke the target.
+///
+/// This InterfaceLinkDefinition is *distinct* from the one in `wasmcloud_core`, in that it is
+/// partially resolved with information related to the required named configuration for a given link
 #[derive(Clone, Debug, Default, Deserialize, PartialEq, Eq, Serialize, Hash)]
-pub struct InterfaceLinkDefinition {
+pub struct ResolvedLinkDefinition {
```

Comment:
I think this one should be the "regular" `Link`, and then the one in `core` should be the `ResolvedLink` because it has the configuration resolved. Does that match up with what you were thinking?

Reply:
OH I see what you mean now -- I was thinking the resolution here was getting the required named config, but you're right, the one in `core` actually has just the KV